### ISSUE TITLE
UrlBuilder should support Form Style Multi-Valued Request Parameters

### ISSUE THREAD

**Issue by lhankins** [RestExpress/HyperExpress#11]
There's not a convenient way to use UrlBuilder / LinkBuilder to create a URL in this form (passing two values for "accountId") :

http://someserver/myapp/report/1234?accountId=400&accountId=401

This is the default format for URLs generated form HTML forms which submit multiple values for the same request parameter (e.g. multiple checkboxes with the same "name" field, a multi-select select box, etc.)

After looking at this issue, I'd suggest two course of action: 

1. TokenResolver should really be an interface - LinkBuilder & UrlBuilder should leverage the interface (not the concrete implementation).   This would enable downstream consumers to provide their own TokenResolver implementations. 
1. The default implementation of the TokenResolver interface ("DefaultTokenResolver") should provide a bind or bindMultiple method which accepts a collection as the value argument.  This would expand the token into repeating value patterns (one for each value).   For example {accountId} gets translated to : accountId=400&accountId=401

Here's a representative unit test which shows both of the above suggestions (assuming TokenResolver is now an interface and DefaultTokenResolver is a concrete implementation) : 

 ```java
   @Test
   public void testLinkBuilder_multiValue()
   {
      String expectedUrl = "http://someserver/myapp/report/1234?accountId=400&accountId=401";

      //--- the list of IDs would be variable...
      List<String> accountIds = Arrays.asList("400", "401");

      Link link = new LinkBuilder("/myapp/report/{reportId}")
            .baseUrl("http://someserver")
            .withQuery("accountId={accountId}")
            .build(new DefaultTokenResolver()
                         .bind("reportId", "1234")
                         .bind("accountId", accountIds)
            );

      assertEquals(expectedUrl, link.getHref());
   }
```

**Comment by tfredrich** [comment]
Splitting course of action 1 into a separate issue: Issue #14

**Comment by tfredrich** [comment]
Available in 2.4-SNAPSHOT
